### PR TITLE
add bitlayer fees adapter

### PR DESCRIPTION
**Pull request by stromman**
we creat a adapter for bitlayer project
PLS check and fallback for fees/bitlayer/index.ts

### CODE REVIEW

**Comment by dtmkeng** [comment]
could you resolve conflicts

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

File: fees/bitlayer/index.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    },
+  });
+  const feesToday = dailyFees
+    .split("\n")
+    .find((d: any) => d?.split(",")?.[1]?.slice(1, -1) == startOfDay);
+  return Number(feesToday?.split(",")[2].slice(1, -2));
+}
+
+const adapter: Adapter = {
+  adapter: {
```
Comment (dtmkeng) [inline fees/bitlayer/index.ts:42]: add `version: 2` above `adapter`

File: fees/bitlayer/index.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          options,
+          "https://api.btrscan.com/scan/v1/chain/txForDefillama"
+        );
+
+        const feeUSD = await getFeeUSD(
+          options,
+          "https://api.btrscan.com/scan/v1/chain/txForDefillama"
+        );
+
+        console.log("amount", amount);
```
Comment (dtmkeng) [inline fees/bitlayer/index.ts]: remove console log

File: fees/bitlayer/index.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        console.log("feeUSD", feeUSD);
+        const dailyFees = feeUSD.toString();
+
+        return {
+          timestamp: options.startOfDay,
+          dailyFees,
+        };
+      },
```
Comment (dtmkeng) [inline fees/bitlayer/index.ts:55]: add `runAtCurrTime: true` above `start`

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    TypeError: Cannot destructure property 'startOfDay' of 'undefined' as it is undefined.
        at getEtherscanFees (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:40:5)
        at fetch (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:68:30)
        at getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:31:25)
        at async Promise.all (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:75:21 {
      chain: 'bitlayer'
    }

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    TypeError: Cannot destructure property 'startOfDay' of 'undefined' as it is undefined.
        at getEtherscanFees (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:40:5)
        at fetch (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:68:30)
        at getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:31:25)
        at async Promise.all (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:75:21 {
      chain: 'bitlayer'
    }

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    TypeError: Cannot destructure property 'startOfDay' of 'undefined' as it is undefined.
        at getEtherscanFees (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:40:5)
        at fetch (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:68:30)
        at getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:31:25)
        at async Promise.all (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:75:21 {
      chain: 'bitlayer'
    }

File: fees/bitlayer/index.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+const adapter: Adapter = {
+  version:2,
+  adapter: {
+    [CHAIN.BITLAYER]: {
+      fetch: async (
+        _timestamp: number,
+        _: ChainBlocks,
+        options: FetchOptions
```
Comment (dtmkeng) [inline fees/bitlayer/index.ts]: param should only (options: FetchOptions)

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    [Error: Request failed with status code 429] {
      url: 'https://api.btrscan.com/scan/v1/chain/txForDefillama',
      method: 'POST',
      axiosError: 'Too Many Requests',
      chain: 'bitlayer'
    }

**Comment by dtmkeng** [comment]
seem api error

**Comment by stromman** [comment]
> seem api error
Frequency Limit

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    [Error: Request failed with status code 429] {
      url: 'https://api.btrscan.com/scan/v1/chain/txForDefillama',
      method: 'POST',
      axiosError: 'Too Many Requests',
      chain: 'bitlayer'
    }

**Comment by dtmkeng** [comment]
still error could you check

**Comment by llamatester** [comment]
The bitlayer.ts adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    BITLAYER 👇
    Backfill start time: 14/4/2024
    NO METHODOLOGY SPECIFIED
    Timestamp: 1715094481 (2024-05-07T15:08:01.000Z)
    Daily fees: 0

**Comment by dtmkeng** [comment]
is there probby some day return daily fees is 0  ?

**Comment by stromman** [comment]
> is there probby some day return daily fees is 0 ?

nop, something wrong.repairing

**Comment by AoWangg** [comment]
@dtmkeng Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out.

**Comment by dtmkeng** [comment]
> @dtmkeng Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out. Thank you very much.

it is this web https://www.bitlayer.org/gas-swap ?

**Comment by stromman** [comment]
> Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out. Thank you very much.

I thank that you mean : https://www.btrscan.com/

**Comment by AoWangg** [comment]
> > @dtmkeng Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out. Thank you very much.
> 
> it is this web https://www.bitlayer.org/gas-swap ?

Thanks for the merge.

Here is some information about Bitlayer

**Website:** https://www.bitlayer.org/
**Twitter:** https://twitter.com/BitLayerLabs

**Comment by llamatester** [comment]
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 9/5/2024
    _______________________________________
    
    BITLAYER 👇
    Backfill start time: 14/4/2024
    NO METHODOLOGY SPECIFIED
    Timestamp: 1715262340 (2024-05-09T13:45:40.000Z)
    Daily fees: 5.37 k

**Comment by dtmkeng** [comment]
hmm why i call at my local it error 
```
[Error: connect ETIMEDOUT 2606:4700:20::681a:4ac:443] {
  url: 'https://api.btrscan.com/scan/v1/chain/txForDefillama',
  method: 'POST',
  chain: 'bitlayer'
}
```

**Pull request merged**